content/slice/encoding/user-defined-types.md

```diff
@@ -93,10 +97,37 @@ enum Fruit : uint16 { Apple, Strawberry, Orange = 300 }
 
 A `Strawberry` is encoded like a `uint16` with value 1 (on 2 bytes). An `Orange` is encoded like a `uint16` with value
 300.
-{% /slice2 %}
 
 The encoding is the same for checked and unchecked enums.
 
+### Enum with fields
+
+An enumerator is encoded as:
+
+- a `varint32` holding its discriminant value
+- for an unchecked enum, a `varuint62` holding the size of the encoded fields
+- a struct holding the fields of this enumerator
+
+The trailing struct is a compact struct when the enum itself is compact; otherwise, it's not compact and its encoding
+always ends with the tag end marker byte.
```

Comment:
I would split this into two sentences and change the wording a bit. It gave me a bit of a hard time the first time reading it.

```suggestion
The trailing struct is a compact struct when the enum itself is compact. Conversely, if the enum is not compact then the trailing struct is also not compact and its encoding always ends with the tag end marker byte.
```